packages/optimizely-sdk/lib/core/optimizely_config/index.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var fns = require('../../utils/fns');
+
+// Get Experiment Ids which are part of rollouts
+function getRolloutExperimentIds(rollouts) {
+  return rollouts.reduce(function(experimentIds, rollout) {
+    return experimentIds.concat(rollout.experiments.map(function(e) {
+      return e.id;
+    }));
+  }, []);
+}
+
+// Gets Map of all experiments except rollouts
+function getExperimentsMap(configObj) {
+  var rolloutExperimentsIds = getRolloutExperimentIds(configObj.rollouts);
+  return configObj.experiments.reduce(function(experiments, experiment) {
+    // skip experiments that are part of a rollout
+    if (!rolloutExperimentsIds.includes(experiment.id)) {
+      experiments[experiment.id] = {
+        id: experiment.id,
+        key: experiment.key,
+        variationsMap: experiment.variations.reduce(function(variations, variation) {
+          variations[variation.id] = {
+            id: variation.id,
+            key: variation.key,
+            variablesMap: variation.variables.reduce(function(variables, variable) {
+              // return empty variable map when feature is disabled
+              if (!variation.featureEnabled) {
+                return {};
+              }
+              variables[variable.id] = {
+                id: variable.id,
+                value: variable.value,
+              };
+              return variables;                
+            }, {}),
+          };
+          return variations;
+        }, {}),
+      };
+    }
+    return experiments;
+  }, {});
+}
+
+// Merges feature varibles in variations of passed in experiment
+// Modifies experiment object.
+function mergeFeatureVariables(experiment, featureVariables) {
+  var variations = fns.values(experiment.variationsMap);
+  variations.forEach(function(variation) {
+    featureVariables.forEach(function(featureVariable) {
+      var variable = variation.variablesMap[featureVariable.id];
```

Comment:
Nit: suggest renaming this
```suggestion
      var variationVariable = variation.variablesMap[featureVariable.id];
```
To clearly distinguish this from `featureVariable`.

Reply:
Done